snappiershot/serializers/utils.py

```diff
@@ -152,5 +153,29 @@ def is_instanced_object(value: Any) -> bool:
     """ Check if the input value is an instanced object, i.e. an instantiated class. """
     is_type = inspect.isclass(value)
     is_function = inspect.isroutine(value)
-    is_object = hasattr(value, "__dict__")
+    is_object = hasattr(value, "__dict__") or hasattr(value, "__slots__")
     return is_object and not is_type and not is_function
+
+
+def fullvars(value: Any) -> dict:
+    """ Returns a mapping of all attributes to their associated values for a given object.
+    Supports slots-optimized classes.
+
+    Args:
+        value: An instantiated class.
+    Returns:
+        A full `vars` dictionary.
+    """
+    # Start with the contents of __dict__ (if it exists).
+    obj_dict = copy(getattr(value, "__dict__", dict()))
+
+    # Iterate through the __slots__ (if they exist).
+    for slot in getattr(value, "__slots__", tuple()):
+        # The __dict__ object may be included in the __slots__.
+        if slot in obj_dict or slot == "__dict__":
+            continue
+        # Slots attributes are not necessarily instantiated.
+        if hasattr(value, slot):
+            obj_dict[slot] = getattr(value, slot)
```

Comment:
do we want a default/fallback value in the `getattr` here?

Reply:
nah, no need to since entering this requires the attribute to exist